src/modules/common/containers/LanguageSelector.js

```diff
@@ -1,37 +1,39 @@
 import React from 'react'
-import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
 
 import LanguageModel from 'modules/endpoint/models/LanguageModel'
 import SelectorItemModel from '../models/SelectorItemModel'
 import Selector from '../components/Selector'
 import CategoriesMapModel from '../../endpoint/models/CategoriesMapModel'
-import { CATEGORIES_ROUTE, goToCategories } from '../../app/routes/categories'
-import { EVENTS_ROUTE, goToEvents } from '../../app/routes/events'
-import { EXTRAS_ROUTE, goToExtras } from '../../app/routes/extras'
-import { DISCLAIMER_ROUTE, goToDisclaimer } from '../../app/routes/disclaimer'
-import { goToSearch, SEARCH_ROUTE } from '../../app/routes/search'
+import { CATEGORIES_ROUTE, getCategoryPath } from '../../app/routes/categories'
+import { EVENTS_ROUTE, getEventPath } from '../../app/routes/events'
+import { EXTRAS_ROUTE, getExtraPath } from '../../app/routes/extras'
+import { DISCLAIMER_ROUTE, getDisclaimerPath } from '../../app/routes/disclaimer'
+import { getSearchPath, SEARCH_ROUTE } from '../../app/routes/search'
 import EventModel from '../../endpoint/models/EventModel'
 import HeaderLanguageSelectorItem from '../../layout/components/HeaderLanguageSelectorItem'
 
+import type { Location } from 'redux-first-router/dist/flow-types'
+import type { State } from '../../../flowTypes'
+
+type Props = {
+  languages: Array<LanguageModel>,
+  location: Location,
+  categories: CategoriesMapModel,
+  events: Array<EventModel>,
+  isHeaderActionItem: boolean
+}
+
 /**
  * Displays a dropDown menu to handle changing of the language
  */
-export class LanguageSelector extends React.Component {
-  static propTypes = {
-    languages: PropTypes.arrayOf(PropTypes.instanceOf(LanguageModel)),
-    location: PropTypes.object.isRequired,
-    categories: PropTypes.instanceOf(CategoriesMapModel),
-    events: PropTypes.arrayOf(PropTypes.instanceOf(EventModel)),
-    isHeaderActionItem: PropTypes.bool.isRequired
-  }
+export class LanguageSelector extends React.Component<Props> {
   /**
    * Maps the given languageCode to an action to go to the current route in the language specified by languageCode
-   * @param languageCode
-   * @return {*}
    */
-  getLanguageChangeAction (languageCode) {
-    const {location, categories, events} = this.props
+  static getLanguageChangePath (params: {| location: Location, categories: CategoriesMapModel,
```

Comment:
I'm not happy, that this method is in the LanguageSelector. This should be moved somewhere else...

Reply:
@Schedulaar do you have a suggestion where to put it? Create a separate file called getLanguageChangePath?

Reply:
Hmm... I'd put it somewhere near the definition of our routes, so probably in the /app module 🤔